experimental/WreathMacdonaldpols/src/WreathMacdonaldpols.jl

```diff
@@ -0,0 +1,265 @@
+end
+
+#Inspired by Sagemath's algorithm to obtain a partition from core and quotient
+
+function tau_om(lbb::Multipartition, wperm::PermGroupElem, coroot::Vector{Integer})
+   r=length(lbb)
+   w0=perm([r-i for i in 0:(r-1)])
+   wperm=inv_perm(wperm,r)
+   lbb_perm=multipartition(permuted(lbb.mp,wperm*w0))
+   gamma=core([-x for x in coroot],r)
+   lg=length(gamma)
+   k=r*maximum(length(lbb_perm[i]) for i in 1:r) + lg
+   v=[[gamma[i]-(i-1) for i in 1:lg]; [-i for i in lg:(k-1)]]
+   w=[[x for x in v if mod((x-i),r) == 0] for i in 1:r]
+   new_w=[]
+   for i in 1:r
+     lw=length(w[i])
+     lq=length(lbb_perm[i])
+     new_w=[new_w;[w[i][j] + r*lbb_perm[i][j] for j in 1:lq]]
+     new_w=[new_w;[w[i][j] for j in lq+1:lw]]
+   end
+   sort!(new_w,rev=true)
+   new_w=[new_w[i]+(i-1) for i in 1:length(new_w)]
+   filter!(x-> x!=0,new_w)
+   return partition(new_w)
+end
+
+#Coroots are given in the canonical basis (e_i) of Z^I.
+#Note also that \alpha^{\vee}_i=e_{i-1}-e_i.
+#The coroot lattice is thus the lattice of null sum elements in Z^I.
+
+function chev_to_osc(charTirr::Matrix{Cyc{Int64}}, r::Integer, K::AbsSimpleNumField)
+  l=size(charTirr)[1]
+  res=AbsSimpleNumFieldElem[]
+  a=gen(K)
+  for j in 1:l
+    for i in 1:l
+      if r > 2
+        coeff=CyclotomicNumbers.coefficients(charTirr[i,j])
+        temp=K(0)
+        for k in 1:length(coeff)
+          temp+=coeff[k]*a^(k-1)
+        end
+        push!(res,temp)
+      else
+        push!(res,K(charTirr[i,j]))
+      end
+    end
+  end
+  return reshape(res,l,l)
+end
+
+#Reoders the CharTable from Chevie
+function reorder(charTirr::Matrix{Cyc{Int64}}, r::Integer, n::Integer, Modules::Vector{Vector{Vector{Vector{Int64}}}}, K::AbsSimpleNumField)
+  mps=multipartitions(n,r)
+  new_ord=Int64[]
+    for lbb in mps
+      i=findfirst(x-> x==lbb,Modules)
+      push!(new_ord,i)
+    end
+  return chev_to_osc(charTirr,r,K)[new_ord,:]
+end
+
+#Tools from representation theory
+
+# Computes the fake degree of a multipartition cf. Ste89 Thm 5.3 and Prop. 3.3.2 Haiman cdm
+function fake_deg(lbb::multipartition, Q::AbstractAlgebra.Generic.FracField{AbstractAlgebra.Generic.MPoly{AbsSimpleNumFieldElem}}, var::AbstractAlgebra.Generic.MPoly{AbsSimpleNumFieldElem})
+  r=size(lbb)[1]
+  n=sum(lbb)[1]
+  res=Q(1)
+  for p in 1:n
+    res=res*(1-var^(r*p))
+  end
+  for k in 1:r
+    for i in 1:length(lbb[k])
+      for j in 1:lbb[k][i]
+        hookfactor=(Q(1)-var^(r*(1+lbb[k][i]+length(findall(row->row >= j,lbb[k]))-j-i)))
+        res=res//hookfactor
+      end
+    end
+  end
+  res=res*var^btot(lbb)
+  return res
+end
+
+# computes C_Delta defined in PhD relation (1.45)
+function C_Delta(r::Integer, n::Integer, K::AbsSimpleNumField, Q::AbstractAlgebra.Generic.FracField{AbstractAlgebra.Generic.MPoly{AbsSimpleNumFieldElem}}, var::AbstractAlgebra.Generic.MPoly{AbsSimpleNumFieldElem})
+  G=complex_reflection_group(r,1,n)
+  Modules=charinfo(G).charparams
+  charTable=CharTable(G)
```

Comment:
There is already `Oscar.character_table_complex_reflection_group(r, 1, n)`,
and one can ask for the `character_parameters` value of that table.
(If this function gets used, I can export the function and add a docstring for it.)